### DISCUSSION TITLE
Some questions about the flux_in_box function

### DISCUSSION THREAD

**Discussion by tmteryz** [NanoComp/meep#2478] [Q&A] [answered]
hi all.
What is the difference between the flux_in_box function and the output_poynting function?
what is the return value of the flux_in_box function ? the instantaneous Poynting flux or Poynting flux spectra?

**Comment by oskooi** [comment, marked as answer]
`flux_in_box` is the *instantaneous* flux computed using the time-domain fields as $\Re\left[\vec{E}^*(t) \times \vec{H}(t)\right]$.

`output_poynting` is also the instantaneous flux but over the entire cell and written to disk as an HDF5 file. 

Note that since the electric and magnetic fields are computed at different times (offset by $\Delta t / 2$), the fields need to be synchronized first to ensure second-order accuracy. This procedure is described in https://meep.readthedocs.io/en/latest/Synchronizing_the_Magnetic_and_Electric_Fields/.

In `flux_in_box`, the fields are automatically synchronized [internally](https://github.com/NanoComp/meep/blob/master/src/energy_and_flux.cpp#L216). However, `output_poynting` requires that you wrap this step function in `synchornized_magnetic` as described in the docs.

**Reply by tmteryz** [reply]
thank you~

**Reply by tmteryz** [reply]
A follow-up question：are the fields  linearly interpolated to the points at the centers of the Yee cells internally in In flux_in_box as in output_poynting?

**Reply by oskooi** [reply]
All fields and field-derived quantities such as the Poynting flux are interpolated to the *center* of the Yee grid voxel for any of the output routines. This is a core feature of Meep referred to as the [Illusion of Continuity](https://meep.readthedocs.io/en/latest/Introduction/#the-illusion-of-continuity).

For more information, see Section 3 "Interpolation and the illusion of continuity" of the [Meep paper](http://ab-initio.mit.edu/~oskooi/papers/Oskooi10.pdf).
